app/src/main/kotlin/edu/stanford/bdh/engagehf/health/bloodpressure/bottomsheet/TimePickerStateMapper.kt

```diff
@@ -0,0 +1,54 @@
+package edu.stanford.bdh.engagehf.health.bloodpressure.bottomsheet
+
+import edu.stanford.spezi.core.utils.DateFormat
+import edu.stanford.spezi.core.utils.DateFormatter
+import edu.stanford.spezi.core.utils.TimeProvider
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.LocalTime
+import java.time.ZoneId
+import javax.inject.Inject
+
+class TimePickerStateMapper @Inject constructor(
+    private val dateFormatter: DateFormatter,
+    private val timeProvider: TimeProvider,
+) {
+
+    fun mapNow(): TimePickerState {
+        val now = timeProvider.nowInstant()
+        val localTime = timeProvider.nowLocalTime()
+        return TimePickerState(
+            selectedDate = now,
+            selectedTime = localTime,
```

Comment:
Instant already has the time, right? Why do we have time and date separate then?

Reply:
We create the time from two ui components, a date picker which sends utc millis at the start of the day, and a time picker which returns the hour and minutes. Then we build the ultimative timestamp from the date of the instant and the hour and minutes of the local time